### PR TITLE
Update personal collection name when user name is updated

### LINKED ISSUE #9863: Personal collection name does not change when changing user name
Metabase version 0.32.5, default H2 database. Current versions of Chrome and IE. Windows 10 build 1809.

I changed a user's name while logged in as an administrator in the Admin settings | People | Edit User.

The changes did take effect: when I go to All Personal collections, I see that user’s collection with the new name. However if I click on it, below “Our Analytics” it displays "OLD USERNAME Personal Collection"

I logged out and logged in as the user and the same results (old name) appear.

### PR DESCRIPTION
**Pull request by lins05**
This shall fix #9863

### CODE REVIEW

**Comment by lins05** [comment]
ping @camsaul . Any chance to take a look?

**Comment by mazameli** [comment]
Sorry for the delay, @lins05. We'll see if we can look at this today or in the next few days. Thanks for the contribution!

**Comment by lins05** [comment]
Ping again :D

**Comment by mazameli** [comment]
@walterl, @flyingmachine, or @sbelak — mind giving this a look?

File: test/metabase/api/user_test.clj [hunk 1]
```diff
@@ -326,9 +329,20 @@
 ;;; |                                      Updating a User -- PUT /api/user/:id                                      |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(defn include-personal-collection-name
```
Comment (walterl) [inline test/metabase/api/user_test.clj]: Is this function being used anywhere?

File: test/metabase/api/user_test.clj [hunk 2]
```diff
@@ -338,13 +352,21 @@
...
-                                     :last_name    "Era"
-                                     :email        "cam.era@metabase.com"
-                                     :is_superuser true}]
-    (let [user (fn [] (into {} (-> (db/select-one [User :first_name :last_name :is_superuser :email], :id user-id)
-                                   (dissoc :common_name))))]
+   :after    {:first_name "Cam"
+              :last_name "Eron"
+              :is_superuser true
+              :email "cam.eron@metabase.com"
+              :personal_collection_name "Cam Eron's Personal Collection"
```
Comment (walterl) [inline test/metabase/api/user_test.clj]: Same closing curly brace issue as above.

File: test/metabase/api/user_test.clj [hunk 3]
```diff
@@ -338,13 +352,21 @@
...
+              }}
+  (tt/with-temp* [User [{user-id :id} {:first_name   "Cam"
+                                       :last_name    "Era"
+                                       :email        "cam.era@metabase.com"
+                                       :is_superuser true}]
+                  Collection [coll]]
+    (let [user (fn [] (into {} (-> (db/select-one [User :id :first_name :last_name :is_superuser :email], :id user-id)
+                                   (hydrate :personal_collection_id)
+                                   (hydrate :personal_collection_name)
```
Comment (walterl) [inline test/metabase/api/user_test.clj]: These two `hydrate`s can be combined into a single step:

```clojure
(hydrate :personal_collection_id :personal_collection_name)
```

**Review by walterl — CHANGES_REQUESTED** [review]
Thanks a bunch for taking the time to do this.

Functionally it looks great! I just have a few style/cosmetic suggestions.

File: test/metabase/api/user_test.clj (hunk 1 above)
Comment (lins05) [reply test/metabase/api/user_test.clj]: Yeah, it's tagged as the hydrate function for `:personal_collection_name` which is used below to verify the name of the personal collection would be changed after updating the user.

```clojure
(defn include-personal-collection-name
  {:hydrate :personal_collection_name}
  ...
```

File: test/metabase/api/user_test.clj [hunk 4]
```diff
@@ -326,9 +329,19 @@
 ;;; |                                      Updating a User -- PUT /api/user/:id                                      |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(defn include-personal-collection-name
+  {:hydrate :personal_collection_name}
+  [user]
+  (db/select-one-field
+      :name Collection :id (:personal_collection_id user)))
```
Comment (walterl) [inline test/metabase/api/user_test.clj]: According to the [Clojure Style Guide](https://guide.clojure.style/#one-space-indent) this line should be indented with one space only. However, in this case I would suggest putting the arguments on the same line as the function call, consistent with how it's used in the rest of the file.

Sorry I missed this before. This is the last change, I'm sure! :stuck_out_tongue_winking_eye:

**Review by walterl — CHANGES_REQUESTED** [review]
One last little change.

**Review by walterl — APPROVED** [review]
Everything looks great! Thanks again for this contribution.

**Comment by lins05** [comment]
Thanks @walterl !

**Pull request merged**